tools/roslaunch/src/roslaunch/rlutil.py

```diff
@@ -207,6 +207,10 @@ def check_roslaunch(f):
         file_deps = {}
     for pkg, miss in missing.items():
         if miss:
+            print("Missing package dependencies: %s/package.xml: %s"%(pkg, ', '.join(miss)))
+        # even if the pkgs is not found in packges.xml, if other package.xml provdes that pkgs, then it will be ok                                                                                           
+        miss_all = list(set(miss) - set(reduce(lambda x, y: x+y, [file_deps[pkg].pkgs for pkg in file_deps.keys()])))
```

Comment:
ok, fixed the logic; https://github.com/ros/ros_comm/pull/730/files#diff-b2e93710e71874c9ea655b6b0644f4c5R210